### ISSUE TITLE
Bug: Content Sync changes empty strings to "\0"

### ISSUE THREAD

**Issue by HitmanInWis** [aemide/issue-tracker#3]
On sync of XML files from CRX where a property is an empty string "", it is synced locally to a "\0" string.

e.g. `value=""` becomes `value="\0"`

**Comment by HitmanInWis** [comment]
@karollewandowski were you ever able to debug this issue?  the `\0` issue is definitely one challenge I'm facing on a Windows machine, but an even bigger issue is the one I mentioned above where it seems the plugin is printing the file contents with unix style line feeds which makes my windows machine think every line of the file has changed from the code repository version.

**Comment by karollewandowski** [comment]
@HitmanInWis, not yet. I'll investigate it as the next task.

If you could also check this:
> Also, could you please try downloading a package with such a problematic node, unpack it somewhere and add a file with an empty property value to the project, and see if it is \0? In this case, the problem would be on the AEM side, not the plugin.

We have Windows users, and nobody reported this issue, so it's a bit suspicious.

**Comment by HitmanInWis** [comment]
> If you could also check this:
> 
> >

I'm replicating this by:
- Create a folder in my project within IntelliJ at jcr_root/libs/core/wcm/components/carousel/v1/carousel
- Pull from Author
- View the downloaded `_cq_dialog/.content.xml` file:

```
                                            <accessibilityPlay
                                                jcr:primaryType="nt:unstructured"
                                                sling:resourceType="granite/ui/components/coral/foundation/form/textfield"
                                                fieldDescription="Value of an aria-label attribute for the carousel navigation's play button label."
                                                fieldLabel="Play"
                                                name="./accessibilityPlay"
                                                value="\0"/>
                                            <accessibilityPause
                                                jcr:primaryType="nt:unstructured"
                                                sling:resourceType="granite/ui/components/coral/foundation/form/textfield"
                                                fieldDescription="Value of an aria-label attribute for the carousel navigation's pause button label."
                                                fieldLabel="Pause"
                                                name="./accessibilityPause"
                                                value="\0"/>
                                            <accessibilityTablist
                                                jcr:primaryType="nt:unstructured"
                                                sling:resourceType="granite/ui/components/coral/foundation/form/textfield"
                                                fieldDescription="Value of an aria-label attribute for the carousel navigation's list of items label."
                                                fieldLabel="Tablist"
                                                name="./accessibilityTablist"
                                                value="\0"/>
```

**Comment by karollewandowski** [comment]
I understand this is pulled with the plugin. I mean testing a package with building it and downloading manually, directly on AEM, without AEM IDE.

AEM IDE uses the package manager under the hood, and if the same issue occurs when you build and download the package with this node manually, we can be sure that it is your AEM instance configuration issue, not the plugin.

**Comment by HitmanInWis** [comment]
Ahh good call, @karollewandowski.  I built a package in package manager, unzipped the package, and opened it in IntelliJ and I see the same `value="\0"` items.  No idea why it's happening, but this confirms it's not the plugin :)

Should we close this issue then, and open a new one for the Windows vs Unix line feeds?

**Comment by karollewandowski** [comment]
Thank you for checking. I tried to google something about AEM or Sling settings but found nothing useful.

I just thought that maybe you run AEM locally in a Linux container like Docker or in WSL (I am not sure if this is possible - I'm not familiar with Windows) and there is a discrepancy in encoding.

Regarding line feeds, this sounds like a similar problem. I understand that it doesn't happen in the package downloaded without the plugin, correct? If so, it is really strange. Let's keep it in this issue.

**Comment by HitmanInWis** [comment]
I'm running AEM directly with java from the Windows command line, so no unix containers invovlved.

Looking at the docs on https://experienceleague.adobe.com/en/docs/experience-manager-65/content/implementing/developing/devtools/ht-vlttool it seems `vlt` is supposed to convert to CRLF automatically on Windows machines.  I assume the plugin is using vlt behind the scenes?  Is the plugin rewriting the content at all, where the plugin would be writing line endings?

**Comment by karollewandowski** [comment]
No, the plugin does not use `vlt`. It uses the Package Manager mechanism. It downloads the ZIP from AEM, unpacks it and copies files to the project. It rewrites the content only in the scope of potential reformatting or deleting junk properties.

**Comment by HitmanInWis** [comment]
Thanks Karol.  I tested today by pulling a content package from CRX for one of my custom components in my codebase.  When I download the package and copy the files in Windows Explorer over the top of my own files, I see that the line endings from AEM are `LF` and thus showing as changed, similar to what I'm seeing with the plugin.

This being the case, the plugin may need the option to rewrite files to CRLF format for windows users.  As mentioned in `vlt` docs, this is something `vlt` already does.  Though the plugin chooses a different method of pulling the content to the codebase, ideally it would handle this the same as `vlt`.

**Comment by karollewandowski** [comment]
Hi @HitmanInWis, I got back to this and thought that maybe this is a matter of your Git configuration. Could you please see if setting a proper `code.autocrlf` value solves the problem?

**Comment by HitmanInWis** [comment]
I think my git setting `core.autocrlf=true` is correct - even though the plugin pulls files back to my IDE as `LF` and confuses the IDE thinking there are file changes (because the file is normally `CRLF` on my windows machine, when I commit to `git` the file is actually not committed.

Given that a dev team can have folks working on both Windows (CRLF) and Unix (LF) machines at the same time, I think there just needs to be a configuration for this.

**Comment by karollewandowski** [comment]
Hi @HitmanInWis , could you please paste some screenshots that show the issue so that I'm sure I understand it?

**Comment by HitmanInWis** [comment]
Sure thing, here is a result of me pulling a very simple page from AEM back to the code base.  Since I'm on Windows, code files are usually in `CRLF` (which you see on the left of the diff) but the files from AEM are coming as `LF` by default.

![Screenshot 2024-12-09 123935](https://github.com/user-attachments/assets/fe59d8d3-3b7d-4d72-987a-3e9037751e99)

**Comment by HitmanInWis** [comment]
FWIW if I "Shelf" the diff, it actually disappears.  So IntelliJ eventually is smart enough to say "hey, this isn't a real diff" but doesn't seem so when the file is updated from the file system by a plugin or other application.
